### PR TITLE
Add colored hotcues for Traktor Kontrol S2 MK3

### PR DESCRIPTION
**Pull request by goddisignz**
Title already says it. I implemented the Mixxx color palette for the hotcue pads of the Traktor Kontrol S2 MK3.

### CODE REVIEW

File: res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js [hunk 1]
```diff
@@ -343,11 +343,42 @@ TraktorS2MK3.padModeHandler = function (field) {
         // Light LEDs for all enabled hotcues
         for (var i = 1; i <= 8; ++i) {
             var active = engine.getValue(field.group, "hotcue_" + i + "_enabled");
-            TraktorS2MK3.outputHandler(active, field.group, "pad_" + i);
+            var color = engine.getValue(field.group, "hotcue_" + i + "_color");
+            if (active) {
```
Comment (ronso0) [inline res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js:347]:
```suggestion
            if (active) {
                        var color = engine.getValue(field.group, "hotcue_" + i + "_color");
```

File: res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js [hunk 2]
```diff
@@ -406,6 +437,7 @@ TraktorS2MK3.loadTrackHandler = function (field) {
         engine.setValue(field.group, "eject", field.value);
     } else {
         engine.setValue(field.group, "LoadSelectedTrack", field.value);
+        engine.setValue(field.group, "pfl", 1);
     }
```
Comment (Swiftb0y) [inline res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js:441]: this is unrelated

Comment (goddisignz) [reply res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js:441]: Yes, sorry, force of habit. I am always thinking of an option in the settings for enabling PFL on track loading.

File: res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js [hunk 3]
```diff
@@ -343,11 +343,42 @@ TraktorS2MK3.padModeHandler = function (field) {
...
+    YELLOW:  0x16,
+    BLUE:    0x2A,
+    PURPLE:  0x32,
+    PINK:    0x3E,
+    WHITE:   0x46,
+    ORANGE:  0x0E,
+
+};
+
+// Mixxx hotcue colors
```
[the remark this answers is not in the document]
Comment (goddisignz) [reply res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js:369]: Nono, this should be the RGB-Colors, the mixxx palette uses. I tried to find a very close color on the pads for each of the palette colors. It is hard to determine the exact RGB-colors the pads have but it should be close.

**Comment by goddisignz** [comment]
BTW: Good to see you guys are still here. I didn't contribute for a while but I will try to do little things again in the future.

File: res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Holzhaus) [reply res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js:369]: > Nono, this should be the RGB-Colors, the mixxx palette uses. I tried to find a very close color on the pads for each of the palette colors.

This is not how the color mapper works. The RGB colors should resemble the pads as close as possible. The Mixxx palette is unrelated, and can be configured with arbitrary colors anyway. 

> It is hard to determine the exact RGB-colors the pads have but it should be close.

I suppose for most of the colors you can take a screenshot of the Traktor GUI and then use a color picker. I'm pretty sure these colors match the LEDs on the controller. Then you know the exact RGB values of green (saved loops), blue (hot cues), white (grid markers), yellow (load in) and orange (fade in/out). The RGB colors (purple/pink/celeste/red) need to be guessed.

[the remark this answers is not in the document]
Comment (goddisignz) [reply res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js]: I have no windows PC so I cannot run traktor. However, I took photos of every possible pad color and analyzed them in gimp using the color picker with an average filter. The result that I see now for the hotcue colors I use looks pretty nice.

**Review by JoergAtGithub — APPROVED** [review]
I read through the changes, and from my point of view everything looks fine. If nobody has objections, I will merge this.

**Comment by JoergAtGithub** [comment]
- Code looks fine to me
- Changes require ES6, therefore Main is the correct target

**Comment by JoergAtGithub** [comment]
@goddisignz Thank you, for your contribution!

**Comment by daschuer** [comment]
Is this a candidate for back-porting to the stable 2.3 branch es well?

**Comment by JoergAtGithub** [comment]
This will not work on 2.3 as it is, due to ES6 syntax. But it should go into 2.4.
